orchagent/fgnhgorch.cpp

```diff
@@ -1546,9 +1803,37 @@ bool FgNhgOrch::doTaskFgNhg(const KeyOpFieldsValuesTuple & t)
                     SWSS_LOG_WARN("Received unsupported match_mode %s, defaulted to route-based",
                                     fvValue(i).c_str());
                 }
+            } 
+            else if (fvField(i) == "nhg_mode")
+            {
+                if (fvValue(i) == "dynamic-nhg")
+                {
+                    nhg_mode = DYNAMIC_FGNHG;
+                }
+                else if (fvValue(i) != "static-nhg")
+                {
+                    SWSS_LOG_WARN("Received unsupported match_mode %s, defaulted to static-nhg",
+                                    fvValue(i).c_str());
+                }
+            } 
+            else if (fvField(i) == "max_next_hops")
+            {
+                max_next_hops = stoi(fvValue(i));
             }
         }
 
+        // Nexthop-based match_mode is not supported for dynamic_fgnhg
+        if (nhg_mode == DYNAMIC_FGNHG && match_mode == NEXTHOP_BASED) 
```

Comment:
Looks like we only log an error in this case, should we default/override to static FGNHG if this happens?

Reply:
Replaced this combination of nhg_mode and match_mode with match_mode==prefix_based; so this check has been removed now.